### PR TITLE
fix: BDD test reports don't stash after using a test cluster

### PR DESCRIPTION
**Pull request by abayer**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

So let's delete `~/.kube/config`, which just contains the test cluster in it, before we stash anything.

#### Special notes for the reviewer(s)

/assign @dgozalo 

#### Which issue this PR fixes

n/a

### CODE REVIEW

**Comment by abayer** [comment]
/hold

Turns out that wasn't sufficient - it's still stashing to, e.g., `gs://jx-prod-logs/jenkins-x/static-e2e-tests/jenkins-x/jx/gh-pages/16/workspace/source/build/reports/junit.html`.

**Comment by abayer** [comment]
/hold cancel

Ok, I got annoyed at the bucket paths being wrong so went and fixed it while I was here. =)

File: jenkins-x-bdd.yml [hunk 1]
```diff
@@ -152,6 +152,10 @@ pipelineConfig:
                   - --suite-name
                   - "Static_BDD_Tests"
 
+              - name: clear-kubeconfig
```
Comment (hferentschik) [inline jenkins-x-bdd.yml]: what's that good for?

**Comment by hferentschik** [comment]
@abayer rebase needed :-)

File: jenkins-x-bdd.yml (hunk 1 above)
Comment (abayer) [reply jenkins-x-bdd.yml]: It's making sure we're no longer using the test cluster context when we subsequently run `jx step stash`.

**Comment by abayer** [comment]
Rebased, after an initial oopsie where my editor got confused and I inadvertently committed the merge conflict itself. Fixed now. =)

**Pull request merged**